mev-boost-rs/src/service.rs

```diff
@@ -45,21 +47,33 @@ fn parse_url(input: &str) -> Option<Url> {
     }
 }
 
+//TODO: Implement Parse for BlsPublicKey in ethereum_consensus
+fn parse_public_key(input: &str) -> Option<BlsPublicKey> {
+    if input.is_empty() {
+        None
+    } else {
+        Some(BlsPublicKey::default())
+    }
+}
+
 pub struct Service {
     host: Ipv4Addr,
     port: u16,
-    relays: Vec<Url>,
+    relays: Vec<(Url, BlsPublicKey)>,
```

Comment:
at this point I'd move to encapsulate these details behind a `Relay` type

I'd do this:

1) add `relay.rs` along with a `mod relay;` in the `lib.rs`
2) define something like:
```rust
pub struct Relay {
    api: BlindedBlockProviderClient,
    public_key: BlsPublicKey,
}
```
3. parse the config `Strings` as URLs, make a client for each URL, parse the public key from the URL's `hostname` and then use some combo of `hex::decode` and `BlsPublicKey::try_from(_: &[u8])` to get a key; provide the client and the key as inputs to a `Relay::new`